src/renderer/component/video/internal/player.jsx

```diff
@@ -175,23 +216,28 @@ class VideoPlayer extends React.PureComponent {
 
   render() {
     const { mediaType, poster } = this.props;
-    const { hasMetadata, unplayable } = this.state;
+    const { hasMetadata, unplayable, unsupported, fileSource } = this.state;
+    const noFileMessage = 'Waiting for blob.';
     const noMetadataMessage = 'Waiting for metadata.';
     const unplayableMessage = "Sorry, looks like we can't play this file.";
-    const hideMedia = this.playableType() && !hasMetadata && !unplayable;
+    const unsupportedMessage = "Sorry, looks like we can't preview this file.";
+    const isLoadingFile = !fileSource && this.fileType();
+    const isLoadingMetadata = this.playableType() && (!hasMetadata && !unplayable);
+    const isUnplayable = this.playableType() && unplayable;
+    const isUnsupported = !this.supportedType() && !this.playableType() && !this.fileType();
+    const isFile = fileSource && this.fileType();
 
     return (
       <React.Fragment>
-        {['audio', 'application'].indexOf(mediaType) !== -1 &&
-          (!this.playableType() || hasMetadata) &&
-          !unplayable && <Thumbnail src={poster} />}
-        {this.playableType() &&
-          !hasMetadata &&
-          !unplayable && <LoadingScreen status={noMetadataMessage} />}
-        {unplayable && <LoadingScreen status={unplayableMessage} spinner={false} />}
+        {isLoadingFile && <LoadingScreen status={noFileMessage} />}
+        {isLoadingMetadata && <LoadingScreen status={noMetadataMessage} />}
+        {isUnplayable && <LoadingScreen status={unplayableMessage} spinner={false} />}
+        {unsupported ||
+          (isUnsupported && <LoadingScreen status={unsupportedMessage} spinner={false} />)}
```

Comment:
I think this would be easier to follow if we had a `shouldDisplayLoadingScreen()` function or something. 

`const [isLoading, loadingStatus] = this.shouldDisplayLoadingScreen()`

Then we would just need one `LoadingScreen` component